### PR TITLE
merge type could be string that is path of existing manifest file

### LINKED ISSUE #100: Question about merge support for two tasks
I have seen the following comment...

https://github.com/sindresorhus/gulp-rev/pull/77#issuecomment-68076646

...yet am still confused about how to realize my merge situation, or whether it is supported. I have two individual gulp tasks, yet I want both to output to the same manifest file. Here's my code:

``` javascript
gulp.task('styles', function() {
    return gulp.src('css/*.css')
  .pipe(minifycss())
  .pipe(gzip({ append: false }))
  .pipe(rev())
  .pipe(gulp.dest('assets'))
  .pipe(rev.manifest('manifest.json',{
            base: './',
            merge: true // merge with the existing manifest (if one exists) 
        }))
  .pipe(gulp.dest('system/application/config/'));
});

gulp.task('scripts', function() {
    return gulp.src('js/*.js')
  .pipe(uglify())
  .pipe(gzip({ append: false }))
  .pipe(rev())
  .pipe(gulp.dest('assets'))
  .pipe(rev.manifest('manifest.json',{
            base: './',
            merge: true // merge with the existing manifest (if one exists) 
        }))
  .pipe(gulp.dest('system/application/config/'));
});
```

So these are 2 individual tasks that are ran independently and sometimes after each other. I'd like them to output to a single, merged manifest. Note that the manifest is in a different directory from the actual revved files. Both the assets and manifest end up in the correct directory, yet the manifest does not merge, it is overwritten by either task.

Any help is welcome.

### PR DESCRIPTION
**Pull request by Aben**
Fixes #100

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gulp-rev",
-  "version": "5.0.1",
+  "version": "5.0.2",
```
Comment (sindresorhus) [inline package.json]: Don't bump versions in other peoples repos ;)

File: readme.md [hunk 2]
```diff
@@ -58,10 +58,11 @@ Override the `cwd` (current working directory) of the manifest file.
 
 ##### merge
 
-Type: `boolean`  
+Type: `boolean` or `string`
 Default: `false`
 
-Merge existing manifest file.
+Merge existing manifest file. If `string`, path of existing manifest file
```
Comment (sindresorhus) [inline readme.md]: > Merge existing manifest file. Optionally specify a path to the manifest file.

File: readme.md [hunk 3]
```diff
@@ -58,10 +58,11 @@ Override the `cwd` (current working directory) of the manifest file.
 
 ##### merge
 
-Type: `boolean`  
+Type: `boolean` or `string`
```
Comment (sindresorhus) [inline readme.md]: > Type: `boolean`, `string`

File: index.js [hunk 4]
```diff
@@ -22,20 +22,27 @@ function relPath(base, filePath) {
...
-	file.read(opts.path, opts, function (err, manifest) {
-		if (err) {
-			// not found
-			if (err.code === 'ENOENT') {
-				cb(null, new gutil.File(opts));
-			} else {
-				cb(err);
-			}
+	var f = new gutil.File(opts)
+	if(opts.merge) {
```
Comment (sindresorhus) [inline index.js]: `if (opts.merge) {`

File: index.js [hunk 5]
```diff
@@ -22,20 +22,27 @@ function relPath(base, filePath) {
...
+		var p = typeof opts.merge == 'string'? opts.merge : opts.path
```
Comment (sindresorhus) [inline index.js]: `var p = typeof opts.merge == 'string' ? opts.merge : opts.path`

**Comment by sindresorhus** [comment]
@bobthecow Thoughts?

File: index.js (hunk 5 above)
Comment (bobthecow) [inline index.js]: How 'bout 

```js
var p = (typeof opts.merge == 'string') ? opts.merge : opts.path;
```

?

There are too many spaces on the left half of that ternary operator for my liking :)

**Comment by sindresorhus** [comment]
@Aben ping :)

**Comment by Aben** [comment]
@sindresorhus fixed and merged. thank you to correct it.

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gulp-rev",
-  "version": "5.1.0",
+  "version": "5.1.1",
```
Comment (sindresorhus) [inline package.json]: Don't bump version in other people's repos ;)

File: index.js [hunk 7]
```diff
@@ -23,20 +23,26 @@ function relPath(base, filePath) {
...
 
 function getManifestFile(opts, cb) {
-	file.read(opts.path, opts, function (err, manifest) {
-		if (err) {
-			// not found
-			if (err.code === 'ENOENT') {
-				cb(null, new gutil.File(opts));
-			} else {
-				cb(err);
+	var f = new gutil.File(opts);
```
Comment (sindresorhus) [inline index.js]: Why? `file.read` returns a vinyl file.

File: index.js [hunk 8]
```diff
@@ -23,20 +23,26 @@ function relPath(base, filePath) {
...
+	if (opts.merge) {
+		var p = typeof opts.merge === 'string' ? opts.merge : opts.path;
+		file.read(p, opts, function (err, manifest) {
+			if (err) {
+				// not found
```
Comment (sindresorhus) [inline index.js]: moot comment, remove.

File: index.js [hunk 9]
```diff
@@ -23,20 +23,26 @@ function relPath(base, filePath) {
...
+				}
+				return;
 			}
-
-			return;
-		}
-
-		cb(null, manifest);
-	});
+			// append contents to f
```
Comment (sindresorhus) [inline index.js]: moot comment

**Comment by sindresorhus** [comment]
Needs a test.

File: package.json (hunk 6 above)
Comment (Aben) [inline package.json]: So, 5.2.0?

Comment (arthurvr) [inline package.json]: No, just don't change this at all. We'll release it but that's separate from this PR.

**Comment by Aben** [comment]
I will read code again.

**Pull request closed without merging**